Websocket telnet fixes

icewolfz opened the pull request:
This removed the extra comment and makes the changes to use maybe_schedule_user_command instead of how the telnet block does it

thefallentree commented:
```
*** CID 1563407:  API usage errors  (PW.TOO_MANY_PRINTF_ARGS)
/home/runner/work/fluffos/fluffos/src/packages/core/telnet_ext.cc: 176 in ()
170     void f_send_msdp_variable() {
171       auto *ip = current_object->interactive;
172       if (ip && ip->telnet) {
173         switch(sp->type) {
174           case T_STRING:
175             telnet_begin_sb(ip->telnet, TELNET_TELOPT_MSDP);
>>>     CID 1563407:  API usage errors  (PW.TOO_MANY_PRINTF_ARGS)
>>>     the format string ends before this argument
176             telnet_printf(ip->telnet, "\x01%s\x02", (sp - 1)->u.string, sp->u.string);
177             telnet_send(ip->telnet, sp->u.string, SVALUE_STRLEN(sp));
178             telnet_finish_sb((ip->telnet));
179             break;
180           case T_NUMBER:
181             telnet_begin_sb(ip->telnet, TELNET_TELOPT_MSDP);


```

got this from the static lint , it does seems 176 is missing another %s ?

thefallentree commented:
actually, it appears that 176 should have sp->u.string deleted

thefallentree commented:
also,  please write something similar to GMCP in docs/apply

thefallentree commented:
reference: https://www.fluffos.info/apply/interactive/gmcp.html#gmcp

icewolfz commented:
i will push a fix i tested it must have hit undo or pushed old file

icewolfz commented:
yeah i see what i did it was a bad file i changed it on my vm testing but forget to pull the fix it should not have the 2nd as i moved that to separate line to avoid printf string limits

icewolfz commented:
ok msdp should be fixed now, i am building a copy to be 100% sure but this push should fix it

icewolfz commented:
I added the docs for all the msdp efuns, and i added the doc for has_msp as well

icewolfz commented:
Okay i think is everything you wanted, added all the docs, fixed the send_msdp_variable, adds the  maybe_schedule_user_command, and removes the extra comment, i tested the fixes on my local copy and i could send msdp with no issues

icewolfz commented:
I updated the testsuite to test MSDP, figured since adding all the docs you would want the testsuite to have the support as well

michaelprograms commented on docs/efun/index.md:
```diff
@@ -199,11 +199,14 @@ title: EFUN
 * [has_mxp](interactive/has_mxp.html)
 * [has_zmp](interactive/has_zmp.html)
+* [has_msp](interactive/has_msp.html)
+* [has_msdp](interactive/has_masdp.html)
```
Shouldn't this be `has_msdp.html`?

icewolfz replied:
yeah it should i miss it, i probably hit the a/s key at the same time and didnt think to double check the link assuming it was right, fixing it now

thefallentree replied:
the index are actually auto generated, just ran `./update_index.sh`

thefallentree commented on docs/apply/interactive/msdp_enable.md:
```diff
@@ -0,0 +1,22 @@
+layout: doc
+title: interactive / msdp_enable
+---
+# gmcp_enable
```
copy/paste error :-D

thefallentree commented on docs/efun/interactive/has_msp.md:
```diff
@@ -0,0 +1,22 @@
+layout: doc
+title: interactive / has_msp
+---
+# has_mxp
```
copy/paste error :-D

icewolfz replied:
Fixed the 3 copy pate errors
